test/pmap_test.rb

```diff
@@ -71,4 +71,11 @@ def test_peach_with_index
 
     assert_equal([["a", 0], ["b", 1], ["c", 2]].sort, output.sort)
   end
+
+  def test_flat_pmap
+    subject = [["a"], [["b"]], [[["c"]]]]
+    proc    = Proc.new { |x| x + ["X"] }
+
```

Comment:
Since the problem is just this test not working on old Rubies which don't have :flat_map, I'd be fine with skipping the test when :flat_map isn't there. Something like this:

    if subject.responds_to?(:flat_map)
      assert_equal(subject.flat_map(&proc), subject.flat_pmap(&proc))
    end